### DISCUSSION TITLE
Give timed revalidation conditional logic support

### DISCUSSION THREAD

**Discussion by CaptShmo** [vercel/next.js#46937] [Ideas]
### Goals

1. I think it would be nice if we could pass a function to revalidate in addition to a time span to make sure revalidation should proceed. Also a function for postProcessing maybe (trigger webhooks, etc). Just an idea, thanks.

### Background

 My use case for this is when I want my CMS to still be live but maybe a lot of content entry is going on and the output needs to be coordinated, so we want to turn off nextjs page revalidation for a while. 

### Proposal

Example:

revalidate: 60, 
checkRevalidate: () => { return true; //or false to cancel revalidation },
onRevalidate: (didRevalidate) => { //do stuff }

**Comment by wwwdepot** [comment]
I think you can use Middleware rewrite to accomplish your goal. AKA ab-testing or feature-flag.

Middleware will fetch the "toggle revalidate" on/off from CMS, depends on the state, rewrite URL to different state or not.

```
/app/page.tsx  <-- revalidate config on
/app/_norevalidate/page.tsx <-- revalidate config off
```

yes, you will have 2 set of pages or more, depends on what you're doing.
yes, this approach is app-wide config, not per-page config.

checkout examples: https://github.com/vercel/examples/tree/main/edge-middleware
